src/AWS.Deploy.Orchestration/TemplateEngine.cs

```diff
@@ -49,50 +42,49 @@ public void GenerateCDKProjectFromTemplate(Recommendation recommendation, Orches
                 recommendation.Recipe.CdkProjectTemplate);
 
             //Installing the base template into the templating engine to make it available for generation
-            InstallTemplates(cdkProjectTemplateDirectory);
+            await InstallTemplates(cdkProjectTemplateDirectory);
 
             //Looking up the installed template in the templating engine
-            var template =
+            var templates = await
                 _bootstrapper
-                    .ListTemplates(
-                        true,
-                        WellKnownSearchFilters.NameFilter(recommendation.Recipe.CdkProjectTemplateId))
-                    .FirstOrDefault()
-                    ?.Info;
+                    .GetTemplatesAsync(
+                        new[] { WellKnownSearchFilters.NameFilter(recommendation.Recipe.CdkProjectTemplateId) });
+            var template = templates.FirstOrDefault()?.Info;
 
             //If the template is not found, throw an exception
             if (template == null)
                 throw new Exception($"Failed to find a Template for [{recommendation.Recipe.CdkProjectTemplateId}]");
 
-            var templateParameters = new Dictionary<string, string> {
-                // CDK Template projects can parameterize the version number of the AWS.Deploy.Recipes.CDK.Common package. This avoid
+            var templateParameters = new Dictionary<string, string?> {
+                // CDK Template projects can parameterize the version number of the AWS.Deploy.Recipes.CDK.Common package. This avoids
                 // projects having to be modified every time the package version is bumped.
                 { "AWSDeployRecipesCDKCommonVersion", FileVersionInfo.GetVersionInfo(typeof(AWS.Deploy.Recipes.CDK.Common.CDKRecipeSetup).Assembly.Location).ProductVersion
                                                       ?? throw new InvalidAWSDeployRecipesCDKCommonVersionException(DeployToolErrorCode.InvalidAWSDeployRecipesCDKCommonVersion, "The version number of the AWS.Deploy.Recipes.CDK.Common package is invalid.") }
             };
 
             try
             {
-                lock (s_locker)
-                {
-                    //Generate the CDK project using the installed template into the output directory
-                    _bootstrapper.CreateAsync(template, assemblyName, outputDirectory, templateParameters, false, "").GetAwaiter().GetResult();
-                }
```

Comment:
why dont we need this lock logic anymore?

Reply:
The locks were there in Microsoft's example's back when we first added them. These days, they've deprecated all the sync functions that used locks and moved to async. Async functions can't be inside locks so I removed them as they shouldnt be needed anymore.